src/metabase/lib/aggregation.cljc

```diff
@@ -226,3 +229,75 @@
                             (-> (lib.metadata.calculation/metadata query stage-number aggregation)
                                 (assoc :lib/source :source/aggregations
                                        ::aggregation-uuid (:lib/uuid (second aggregation))))))))))
+
+(def ^:private OperatorWithColumns
+  [:merge
+   ::lib.schema.aggregation/operator
+   [:map
+    [:columns {:optional true} [:sequential lib.metadata/ColumnMetadata]]]])
+
+(defmethod lib.metadata.calculation/display-name-method :mbql.aggregation/operator
+  [_query _stage-number {:keys [display-info]} _display-name-style]
+  (:display-name (display-info)))
+
+(defmethod lib.metadata.calculation/display-info-method :mbql.aggregation/operator
+  [_query _stage-number {:keys [display-info requires-field?] short-name :short}]
+  (assoc (display-info)
+         :short short-name
+         :requires-field requires-field?))
```

Comment:
Would it make sense to use `requires-column` instead of `requires-field` according to #30060 ?